src/tests/Microsoft.PowerFx.Core.Tests/IntellisenseTests/TokenizationTests.cs

```diff
@@ -0,0 +1,203 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.PowerFx.Core.Binding;
+using Microsoft.PowerFx.Core.Binding.BindInfo;
+using Microsoft.PowerFx.Core.Functions;
+using Microsoft.PowerFx.Core.Texl.Intellisense;
+using Microsoft.PowerFx.Core.Types;
+using Microsoft.PowerFx.Core.Types.Enums;
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Intellisense;
+using Microsoft.PowerFx.Syntax;
+using Xunit;
+
+namespace Microsoft.PowerFx.Core.Tests.IntellisenseTests
+{
+    public class TokenizationTests : PowerFxTest
+    {
+        [Theory]
+        [InlineData("gallery1", 1, 0, "gallery1", 0, 8, TokenType.Control, false)]
+        [InlineData("gallery1!Selected!nestedLabel1", 3, 0, "gallery1", 0, 8, TokenType.Control, false)]
```

Comment:
>! [](http://example.com/codeflow?start=29&length=1)

we're moving away from allowing `!` in any cases, consider just leaving out those tests.

Reply:
Removed those test cases